components/notebook-controller/controllers/notebook_controller.go

```diff
@@ -485,7 +485,12 @@ func generateVirtualService(instance *v1beta1.Notebook) (*unstructured.Unstructu
 	vsvc.SetKind("VirtualService")
 	vsvc.SetName(virtualServiceName(name, namespace))
 	vsvc.SetNamespace(namespace)
-	if err := unstructured.SetNestedStringSlice(vsvc.Object, []string{"*"}, "spec", "hosts"); err != nil {
+
+	istioHosts := os.Getenv("ISTIO_HOSTS")
+	if len(istioHosts) == 0 {
```

Comment:
@vvnpan only one minor nit, could you instead use the `GetEnvDefault` helper we have here for this one?

https://github.com/kubeflow/kubeflow/blob/master/components/notebook-controller/pkg/culler/culler.go#L60

Rest LGTM

Reply:
@kimwnasptd we might have a trade off for the user readability here if we call the `culler. GetEnvDefault` function, it could be confusing seeing a function from the culler package within the `generateVirtualService` function. 
But if that's the structure the maintainers want to follow and break away from the [existing](https://github.com/kubeflow/kubeflow/blob/da6efb9c9fcdfe03a2932aacc6df32489f2883a3/components/notebook-controller/controllers/notebook_controller.go#L492) I will update with the changes